### PR TITLE
Migrate to android x

### PR DESCRIPTION
**Pull request by cozej4**
[Migrate to androidX](https://github.com/ellykits/neat-form/issues/50)

### CODE REVIEW

**Comment by cozej4** [comment]
@ellykits I am working on integrating of  [neat stepper library for multi step forms.](https://github.com/ellykits/neat-form/issues/43). 
Since the library is already migrated to AndroidX, We need to migrate neat form core in order to integrate the two.

File: neat-form-core/src/main/java/com/nerdstone/neatformcore/views/builders/TextInputEditTextBuilder.kt [hunk 1]
```diff
@@ -76,7 +79,7 @@ class TextInputEditTextBuilder(override val nFormView: NFormView) : ViewBuilder
         if (textInputLayout.editText == null)
             textInputLayout.addView(textInputEditTextNFormView, editTextParams);
         else
-            textInputEditTextNFormView = textInputLayout.editText as TextInputEditText;
+            textInputEditTextNFormView = textInputLayout.editText as com.google.android.material.textfield.TextInputEditText;
```
Comment (ellykits) [inline neat-form-core/src/main/java/com/nerdstone/neatformcore/views/builders/TextInputEditTextBuilder.kt]: Redundant qualifier name

File: neat-form-core/src/main/java/com/nerdstone/neatformcore/views/widgets/DateTimePickerNFormView.kt [hunk 2]
```diff
@@ -12,7 +12,7 @@ import com.nerdstone.neatformcore.utils.ViewUtils
 import com.nerdstone.neatformcore.views.builders.DateTimePickerViewBuilder
 import com.nerdstone.neatformcore.views.handlers.ViewDispatcher
 
-class DateTimePickerNFormView : TextInputLayout, NFormView {
+class DateTimePickerNFormView : com.google.android.material.textfield.TextInputLayout, NFormView {
```
Comment (ellykits) [inline neat-form-core/src/main/java/com/nerdstone/neatformcore/views/widgets/DateTimePickerNFormView.kt]: Redundant qualifier name here. Can be optimized

File: neat-form-core/src/main/java/com/nerdstone/neatformcore/views/widgets/TextInputEditTextNFormView.kt [hunk 3]
```diff
@@ -13,7 +13,7 @@ import com.nerdstone.neatformcore.utils.ViewUtils
 import com.nerdstone.neatformcore.views.builders.TextInputEditTextBuilder
 import com.nerdstone.neatformcore.views.handlers.ViewDispatcher
 
-class TextInputEditTextNFormView : TextInputLayout, NFormView {
+class TextInputEditTextNFormView : com.google.android.material.textfield.TextInputLayout, NFormView {
```
Comment (ellykits) [inline neat-form-core/src/main/java/com/nerdstone/neatformcore/views/widgets/TextInputEditTextNFormView.kt]: Redundant qualifier name here. Can be optimized

File: neat-form-core/src/test/java/com/nerdstone/neatformcore/robolectric/builders/JsonFormBuilderTest.kt [hunk 4]
```diff
@@ -88,10 +88,10 @@ class `Test building form with JSON` {
 
                 val verticalRootView = mainLayout.getChildAt(0) as VerticalRootView
                 Assert.assertTrue(verticalRootView.childCount == 1)
-                Assert.assertTrue((verticalRootView.getChildAt(0) as ConstraintLayout).getChildAt(4) is EditTextNFormView)
+                Assert.assertTrue((verticalRootView.getChildAt(0) as androidx.constraintlayout.widget.ConstraintLayout).getChildAt(4) is EditTextNFormView)
```
Comment (ellykits) [inline neat-form-core/src/test/java/com/nerdstone/neatformcore/robolectric/builders/JsonFormBuilderTest.kt]: Redundant qualifier name here. Can be optimized

File: neat-form-core/src/test/java/com/nerdstone/neatformcore/robolectric/builders/JsonFormBuilderTest.kt [hunk 5]
```diff
@@ -88,10 +88,10 @@ class `Test building form with JSON` {
...
 
                 val editTextAttributes =
-                        ((verticalRootView.getChildAt(0) as ConstraintLayout).getChildAt(4) as EditTextNFormView).viewProperties.viewAttributes as Map<*, *>
+                        ((verticalRootView.getChildAt(0) as androidx.constraintlayout.widget.ConstraintLayout).getChildAt(4) as EditTextNFormView).viewProperties.viewAttributes as Map<*, *>
```
Comment (ellykits) [inline neat-form-core/src/test/java/com/nerdstone/neatformcore/robolectric/builders/JsonFormBuilderTest.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/androidTest/java/com/nerdstone/neatform/ExampleInstrumentedTest.kt [hunk 6]
```diff
@@ -11,13 +11,13 @@ import org.junit.runner.RunWith
...
  */
-@RunWith(AndroidJUnit4::class)
+@RunWith(androidx.test.ext.junit.runners.AndroidJUnit4::class)
 class ExampleInstrumentedTest {
 
     @Test
     fun useAppContext() {
         // Context of the app under test.
-        val appContext = InstrumentationRegistry.getTargetContext()
+        val appContext = androidx.test.platform.app.InstrumentationRegistry.getTargetContext()
```
Comment (ellykits) [inline sample/src/androidTest/java/com/nerdstone/neatform/ExampleInstrumentedTest.kt]: Kindly delete this sample test class . We forgot to remove it

File: sample/src/main/java/com/nerdstone/neatform/MainActivity.kt [hunk 7]
```diff
@@ -16,8 +16,8 @@ import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity(), View.OnClickListener {
 
-    private lateinit var formRecyclerView: RecyclerView
-    private lateinit var floatingActionButton: FloatingActionButton
+    private lateinit var formRecyclerView: androidx.recyclerview.widget.RecyclerView
+    private lateinit var floatingActionButton: com.google.android.material.floatingactionbutton.FloatingActionButton
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/MainActivity.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/main/java/com/nerdstone/neatform/MainActivity.kt [hunk 8]
```diff
@@ -51,7 +52,7 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
 
         formRecyclerView.adapter = formRecyclerAdapter
         formRecyclerAdapter.listener = View.OnClickListener {
-            val viewHolder = it.tag as RecyclerView.ViewHolder
+            val viewHolder = it.tag as androidx.recyclerview.widget.RecyclerView.ViewHolder
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/MainActivity.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/main/java/com/nerdstone/neatform/MainActivity.kt [hunk 9]
```diff
@@ -65,10 +66,10 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
 
     override fun onClick(v: View?) {
         when {
-            v?.id == R.id.newFormFab -> Snackbar.make(
+            v?.id == R.id.newFormFab -> com.google.android.material.snackbar.Snackbar.make(
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/MainActivity.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/main/java/com/nerdstone/neatform/MainActivity.kt [hunk 10]
```diff
@@ -65,10 +66,10 @@ class MainActivity : AppCompatActivity(), View.OnClickListener {
...
                 findViewById(R.id.mainActivityConstraintLayout),
                 "Action not yet implemented",
-                Snackbar.LENGTH_SHORT
+                com.google.android.material.snackbar.Snackbar.LENGTH_SHORT
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/MainActivity.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/main/java/com/nerdstone/neatform/form/FormRecyclerAdapter.kt [hunk 11]
```diff
@@ -1,13 +1,13 @@
...
-import android.support.v7.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
 import com.nerdstone.neatform.R
 
-class FormRecyclerAdapter : RecyclerView.Adapter<FormRecyclerAdapter.MainViewHolder>() {
+class FormRecyclerAdapter : androidx.recyclerview.widget.RecyclerView.Adapter<FormRecyclerAdapter.MainViewHolder>() {
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/form/FormRecyclerAdapter.kt]: Redundant qualifier name here. Can be optimized

File: sample/src/main/java/com/nerdstone/neatform/form/FormRecyclerAdapter.kt [hunk 12]
```diff
@@ -32,7 +32,7 @@ class FormRecyclerAdapter : RecyclerView.Adapter<FormRecyclerAdapter.MainViewHol
         }
     }
 
-    inner class MainViewHolder(rowSection: View) : RecyclerView.ViewHolder(rowSection) {
+    inner class MainViewHolder(rowSection: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(rowSection) {
```
Comment (ellykits) [inline sample/src/main/java/com/nerdstone/neatform/form/FormRecyclerAdapter.kt]: Redundant qualifier name here. Can be optimized

File: neat-form-core/src/androidTest/java/com/nerdstone/neatformcore/ExampleInstrumentedTest.kt [hunk 13]
```diff
@@ -1,7 +1,7 @@
 package com.nerdstone.neatformcore
 
-import android.support.test.InstrumentationRegistry
-import android.support.test.runner.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
```
Comment (ellykits) [inline neat-form-core/src/androidTest/java/com/nerdstone/neatformcore/ExampleInstrumentedTest.kt]: Kindly delete

**Pull request merged**